File: Lab/Atv-1/A_Iterativo.c
#include <stdio.h>
#include <string.h>

int isUppercase(char c)
{
      return (c >= 'A' && c <= 'Z');
}

int isEnd(const char *str)
{
      return strcmp(str, "FIM") == 0;
}

int countUppercase(const char *str)
{
      int count = 0;
      for (int i = 0; str[i] != '\0'; i++)
      {
            if (isUppercase(str[i]))
            {
                  count++;
            }
      }
      return count;
}

int main()
{
      char str[100];
      int count;

      do
      {
            fgets(str, sizeof(str), stdin);
            strtok(str, "\n");
            if (!isEnd(str))
            {
                  count = countUppercase(str);
                  printf("%d\n", count);
            }
      } while (!isEnd(str));

      return 0;
}

File: Lab/Atv-1/A_Iterativo.java
import java.util.Scanner;

public class A_Iterativo {
    public static boolean isUppercase(char c) {
        return (c >= 'A' && c <= 'Z');
    }

    public static boolean isEnd(String str) {
        return str.equals("FIM");
    }

    public static int countUppercase(String str) {
        int count = 0;
        for (int i = 0; i < str.length(); i++) {
            if (isUppercase(str.charAt(i))) {
                count++;
            }
        }
        return count;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String str;
        int count;

        do {
            str = scanner.nextLine();
            if (!isEnd(str)) {
                count = countUppercase(str);
                System.out.println(count);
            }
        } while (!isEnd(str));
    }
}